From: Shahed Shaikh <shahed.shaikh@qlogic.com>
Date: Mon, 24 Feb 2014 08:54:46 -0500
Subject: [PATCH 210/270] qlcnic: Fix function return error check
Patch-mainline: v3.14-rc6
Git-commit: fc49beaee2a410402f49fd21c81a37a863b7a9ba
References: bsc#909350 FATE#317546

Driver was treating -ve return value as success in case of
qlcnic_enable_msi_legacy() failure

Signed-off-by: Shahed Shaikh <shahed.shaikh@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Gary Ching-Pang Lin <glin@suse.com>
---
 drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -816,7 +816,7 @@ static int qlcnic_82xx_setup_intr(struct
 			qlcnic_disable_multi_tx(adapter);
 
 			err = qlcnic_enable_msi_legacy(adapter);
-			if (!err)
+			if (err)
 				return err;
 		}
 	}
